webapp/src/index.tsx

```diff
@@ -765,13 +765,24 @@ export default class Plugin {
             return actions;
         };
 
+        // Run onActivate once we're logged in. Only run it once.
+        let activated = false;
+        const unsubscribeActivateListener = store.subscribe(() => {
```

Comment:
It lets us remove the listener after we've logged in. If we added it to the one below, we would be making the check on every store update until the plugin unloads.

Reply:
Mmm, but we are keeping the other listener around anyway. I'd expect to avoid checking again through the `activated` variable.

Reply:
Bit of a tangent, but technically we could remove the `activated` var, but I wanted to be extra sure we didn't run `onActivated` more than needed. Anyway, point being: we avoid the need for any future checks by removing the listener, so why not do it that way?
As an aside: I kind of don't like that store listener below, and hope we can remove it in the future. :)

Reply:
Why don't you like it? It's the only way we have to react to the user switching channels which is when we sync the state of the call. I don't think we can get rid of it completely without implementing an equivalent hook.

> Anyway, point being: we avoid the need for any future checks by removing the listener, so why not do it that way?

It just felt extra complexity to me since we have a listener around already and still have to do the variable check, but not a big deal. Since we are here do you think we could review the `configRetrieved` logic and see if we could simplify?

Reply:
I removed the `activated` var (as I should have from the start), and the `configRetrieved` var is likewise unnecessary since we'll now always be logged in when `onActivate` is run. This doesn't prevent the situation of a failed `/config` fetch. Open to your thoughts on this. We could add retry logic, but I think that might be overkill (and we don't do that in all our other api calls, or the other config call in the standalone). On the other hand, if that call fails, we're kind of operating without a config until a refresh (same as the standalone as of now).

Reply:
I think it's fine for now. What if the re-try fails? Either we design a system that surfaces the error properly (e.g. through the global banner) or it's a bit pointless in my mind. The important thing is that the error shows up in the console which it should in case of debugging.